src/overlays/actors/ovl_En_Fall/z_en_fall.c

```diff
@@ -36,75 +64,820 @@ const ActorInit En_Fall_InitVars = {
+    this->actor.world.rot.x = Math_Vec3f_Pitch(&this->actor.world.pos, &this->actor.home.pos);
+    this->actor.speedXZ = Math_Vec3f_DistXYZ(&this->actor.world.pos, &this->actor.home.pos) / 82.0f;
+    this->actor.shape.rot.x = this->actor.world.rot.x;
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+}
+
+void EnFall_DoNothing(EnFall* this, GlobalContext* globalCtx) {
+}
+
+void EnFall_MoonsTear_Fall(EnFall* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    if (func_800EE29C(globalCtx, 0x205) && globalCtx->csCtx.npcActions[func_800EE200(globalCtx, 0x205)]->unk0 == 2 &&
+        this->actor.draw == NULL) {
+        EnFall_MoonsTear_Initialize(this);
+    }
+
+    if (this->actor.draw != NULL) {
+        if (Math_Vec3f_StepTo(&this->actor.world.pos, &this->actor.home.pos, this->actor.speedXZ) <= 0.0f) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_GORON_BOUND_1);
+            gSaveContext.weekEventReg[0x4A] |= 0x80;
+            gSaveContext.weekEventReg[0x4A] |= 0x20;
+            Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_TEST, this->actor.world.pos.x,
+                               this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, -2);
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG, this->actor.world.pos.x,
+                        this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, CLEAR_TAG_LARGE_EXPLOSION);
+            this->actor.draw = NULL;
+            this->actionFunc = EnFall_DoNothing;
+        } else {
+            func_800B9010(&this->actor, NA_SE_EV_MOONSTONE_FALL - SFX_FLAG);
+        }
+    }
+}
+
+void EnFall_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnFall* this = THIS;
+    this->actionFunc(this, globalCtx);
+}
+
+/**
+ * Updates the alpha for every vertex in the fire ball so that some parts of
+ * the sphere are more transparent or opaque than others.
+ */
+void EnFall_FireBall_SetPerVertexAlpha(f32 fireBallAlpha) {
+    static u8 sAlphaTableIndex[] = { 4, 4, 0, 1, 1, 1, 1, 1, 1, 3, 3, 0, 0, 3, 0, 1, 1, 1, 4, 0, 4, 0, 1, 1, 1, 3, 0,
+                                     3, 0, 1, 1, 1, 4, 4, 1, 1, 0, 4, 4, 0, 1, 1, 1, 1, 1, 1, 3, 3, 0, 0, 3, 3, 0, 0,
+                                     1, 1, 1, 1, 1, 4, 0, 4, 4, 0, 4, 4, 1, 1, 1, 1, 1, 1, 3, 3, 0, 0, 3, 3, 0, 0, 1,
+                                     1, 1, 1, 1, 1, 4, 4, 0, 4, 0, 1, 1, 1, 3, 0, 3, 3, 0, 0, 1, 1, 1, 1, 1, 1, 4, 4,
+                                     0, 4, 4, 0, 1, 1, 1, 1, 1, 1, 3, 3, 0, 0, 0, 0, 0, 2, 2, 1, 1, 0, 0, 2, 2, 0, 1,
+                                     1, 0, 0, 1, 0, 2, 0, 0, 1, 0, 2, 0, 0, 2, 1, 2, 0, 1, 0, 1, 0, 0, 1, 2, 2, 0, 0,
+                                     2, 1, 1, 0, 0, 1, 1, 0, 0, 2, 2, 0, 0, 0, 0, 2, 0, 2, 2, 0, 1, 1, 0, 0, 1, 0, 0,
+                                     1, 2, 2, 0, 0, 0, 2, 2, 1, 1, 0, 0, 1, 1, 0, 0, 2, 2, 0, 0 };
+
+    s32 pad;
```

Comment:
```suggestion
    static u8 sAlphaTableIndex[] = { 4, 4, 0, 1, 1, 1, 1, 1, 1, 3, 3, 0, 0, 3, 0, 1, 1, 1, 4, 0, 4, 0, 1, 1, 1, 3, 0,
                                     3, 0, 1, 1, 1, 4, 4, 1, 1, 0, 4, 4, 0, 1, 1, 1, 1, 1, 1, 3, 3, 0, 0, 3, 3, 0, 0,
                                     1, 1, 1, 1, 1, 4, 0, 4, 4, 0, 4, 4, 1, 1, 1, 1, 1, 1, 3, 3, 0, 0, 3, 3, 0, 0, 1,
                                     1, 1, 1, 1, 1, 4, 4, 0, 4, 0, 1, 1, 1, 3, 0, 3, 3, 0, 0, 1, 1, 1, 1, 1, 1, 4, 4,
                                     0, 4, 4, 0, 1, 1, 1, 1, 1, 1, 3, 3, 0, 0, 0, 0, 0, 2, 2, 1, 1, 0, 0, 2, 2, 0, 1,
                                     1, 0, 0, 1, 0, 2, 0, 0, 1, 0, 2, 0, 0, 2, 1, 2, 0, 1, 0, 1, 0, 0, 1, 2, 2, 0, 0,
                                     2, 1, 1, 0, 0, 1, 1, 0, 0, 2, 2, 0, 0, 0, 0, 2, 0, 2, 2, 0, 1, 1, 0, 0, 1, 0, 0,
                                     1, 2, 2, 0, 0, 0, 2, 2, 1, 1, 0, 0, 1, 1, 0, 0, 2, 2, 0, 0, };

    s32 pad;
```
Have you tried adding a comma to this to make it more compact? Do whichever looks better